plugins/woocommerce/src/Blocks/Domain/Services/CheckoutFieldsFrontend.php

```diff
@@ -0,0 +1,252 @@
+
+			if ( 'select' === $field['type'] ) {
+				$form_field['options'] = array_column( $field['options'], 'label', 'value' );
+			}
+
+			if ( 'checkbox' === $field['type'] ) {
+				$form_field['checked_value']   = '1';
+				$form_field['unchecked_value'] = '0';
+			}
+
+			woocommerce_form_field( $key, $form_field, wc_get_post_data_by_key( $key, $form_field['value'] ) );
+		}
+	}
+
+	/**
+	 * Validates and saves additional address fields to the customer object on the My Account page.
+	 *
+	 * @param integer $user_id User ID.
+	 */
+	public function save_account_form_fields( $user_id ) {
+		$customer = new WC_Customer( $user_id );
+		$fields   = $this->checkout_fields_controller->get_fields_for_location( 'contact' );
+
+		foreach ( $fields as $field_key => $field ) {
+			if ( ! isset( $_POST[ $field_key ] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Missing
+				continue;
+			}
+			$this->checkout_fields_controller->persist_field_for_customer( $field_key, wc_clean( wp_unslash( $_POST[ $field_key ] ) ), $customer );
+		}
+	}
+
+	/**
+	 * Adds additional address fields to the My Account edit address form.
+	 *
+	 * @param array  $address Address fields.
+	 * @param string $address_type Type of address (billing or shipping).
+	 * @return array Updated address fields.
+	 */
+	public function edit_address_fields( $address, $address_type ) {
+		$customer = new WC_Customer( get_current_user_id() );
+		$fields   = $this->checkout_fields_controller->get_fields_for_location( 'address' );
+
+		foreach ( $fields as $key => $field ) {
+			$field_key                      = 'billing' === $address_type ? '/billing/' . $key : '/shipping/' . $key;
+			$address[ $field_key ]          = $field;
+			$address[ $field_key ]['value'] = $this->checkout_fields_controller->get_field_from_customer( $key, $customer, $address_type );
+
+			if ( 'select' === $field['type'] ) {
+				$address[ $field_key ]['options'] = array_column( $field['options'], 'label', 'value' );
+			}
+
+			if ( 'checkbox' === $field['type'] ) {
+				$address[ $field_key ]['checked_value']   = '1';
+				$address[ $field_key ]['unchecked_value'] = '0';
+			}
+		}
+
+		return $address;
+	}
+
+	/**
+	 * Validates and saves additional address fields to the customer object on the My Account page.
+	 *
+	 * @param integer $user_id User ID.
+	 * @param string  $address_type Type of address (billing or shipping).
+	 */
+	public function save_address_fields( $user_id, $address_type ) {
+		$customer = new WC_Customer( $user_id );
+		$fields   = $this->checkout_fields_controller->get_fields_for_location( 'address' );
+
+		foreach ( $fields as $key => $field ) {
+			$field_key = 'billing' === $address_type ? '/billing/' . $key : '/shipping/' . $key;
+
+			if ( ! isset( $_POST[ $field_key ] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Missing
+				continue;
+			}
+
+			// phpcs:ignore WordPress.Security.NonceVerification.Missing
```

Comment:
Can you confirm this is fine because nonce verification took place in the core form handling? (Same for other instances of this)

Reply:
Yeah, we're hooking into things like `woocommerce_after_save_address_validation` where nonce was already checked.